scripts/build_renewable_profiles.py

```diff
@@ -180,220 +180,158 @@
-    feature = gk.vector.extractFeature(paths["regions"], where=gid)
-    ec = gl.ExclusionCalculator(feature.geom)
+    if config['natura']:
+        excluder.add_raster(paths.natura, nodata=0, allow_no_overlap=True)
 
     corine = config.get("corine", {})
-    if isinstance(corine, list):
-        corine = {'grid_codes': corine}
     if "grid_codes" in corine:
-        ec.excludeRasterType(clc, value=corine["grid_codes"], invert=True)
+        codes = corine["grid_codes"]
+        excluder.add_raster(paths.corine, codes=codes, invert=True, crs=3035)
     if corine.get("distance", 0.) > 0.:
-        ec.excludeRasterType(clc, value=corine["distance_grid_codes"], buffer=corine["distance"])
+        codes = corine["distance_grid_codes"]
+        buffer = corine["distance"]
+        excluder.add_raster(paths.corine, codes=codes, buffer=buffer, crs=3035)
 
-    if config.get("natura", False):
-        ec.excludeRasterType(natura, value=1)
     if "max_depth" in config:
-        ec.excludeRasterType(gebco, (None, -config["max_depth"]))
+        func = lambda v: v <= -config['max_depth']
+        excluder.add_raster(paths.gebco, codes=func, crs=4236, nodata=-1000)
 
-    # TODO compute a distance field as a raster beforehand
-    if 'max_shore_distance' in config:
-        ec.excludeVectorType(paths["country_shapes"], buffer=config['max_shore_distance'], invert=True)
     if 'min_shore_distance' in config:
-        ec.excludeVectorType(paths["country_shapes"], buffer=config['min_shore_distance'])
-
-    if save_map is not None:
-        ec.draw()
-        plt.savefig(save_map, transparent=True)
-        plt.close()
-
-    availability = downsample_to_coarse_grid(bounds, dx, dy, ec.region, np.where(ec.region.mask, ec._availability, 0))
-
-    return csr_matrix(gk.raster.extractMatrix(availability).flatten() / 100.)
-
-
-if __name__ == '__main__':
-    if 'snakemake' not in globals():
-        from _helpers import mock_snakemake
-        snakemake = mock_snakemake('build_renewable_profiles', technology='solar')
-    configure_logging(snakemake)
+        buffer = config['min_shore_distance']
+        excluder.add_geometry(paths.country_shapes, buffer=buffer)
 
-    pgb.streams.wrap_stderr()
+    if 'max_shore_distance' in config:
+        buffer = config['max_shore_distance']
+        excluder.add_geometry(paths.country_shapes, buffer=buffer, invert=True)
+
+    kwargs = dict(nprocesses=nprocesses, disable_progressbar=noprogress)
+    if noprogress:
+        logger.info('Calculate landuse availabilities...')
+        start = time.time()
+        availability = cutout.availabilitymatrix(regions, excluder, **kwargs)
+        duration = time.time() - start
+        logger.info(f'Completed availability calculation ({duration:2.2f}s)')
+    else:
+        availability = cutout.availabilitymatrix(regions, excluder, **kwargs)
 
-    config = snakemake.config['renewable'][snakemake.wildcards.technology]
+    area = cutout.grid.to_crs({'proj': 'cea'}).area / 1e6
```

Comment:
I think one could make this tidier with
```suggestion
    area = cutout.grid.to_crs(epsg=xxxx).area / 1e6
```

Reply:
done